server/catgenome/src/main/java/com/epam/catgenome/controller/heatmap/HeatmapController.java

```diff
@@ -0,0 +1,158 @@
+    }
+
+    @GetMapping(value = "/heatmap/{heatmapId}/annotation")
+    @ApiOperation(
+            value = "Returns heatmap annotation for labels",
+            notes = "Returns heatmap annotation for labels",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+            value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+            })
+    public Result<List<List<String>>> getLabelAnnotation(@PathVariable final long heatmapId) throws IOException {
+        return Result.success(heatmapSecurityService.getLabelAnnotation(heatmapId));
+    }
+
+    @PutMapping(value = "/heatmap/{heatmapId}/annotation")
+    @ApiOperation(
+            value = "Updates heatmap annotation for labels",
+            notes = "Updates heatmap annotation for labels",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+            value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+            })
+    public Result<Boolean> updateLabelAnnotation(@PathVariable final long heatmapId,
+                                            @RequestParam final String path) throws IOException {
+        heatmapSecurityService.updateLabelAnnotation(heatmapId, path);
+        return Result.success(null);
+    }
+
+    @PutMapping(value = "/heatmap/{heatmapId}/cell/annotation")
+    @ApiOperation(
+            value = "Updates heatmap annotation for cells",
+            notes = "Updates heatmap annotation for cells",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+            value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+            })
+    public Result<Boolean> updateCellAnnotation(@PathVariable final long heatmapId,
+                                                @RequestParam final String path) throws IOException {
+        heatmapSecurityService.updateCellAnnotation(heatmapId, path);
+        return Result.success(null);
+    }
+
+    @GetMapping(value = "/heatmap/{heatmapId}/tree")
+    @ApiOperation(
+            value = "Returns heatmap tree",
+            notes = "Returns heatmap tree",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+            value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+            })
+    public Result<HeatmapTree> getTree(@PathVariable final long heatmapId) {
+        return Result.success(heatmapSecurityService.getTree(heatmapId));
+    }
+
+    @PostMapping(value = "/heatmap")
+    @ApiOperation(
+            value = "Registers new heatmap",
+            notes = "Registers new heatmap",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+            value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+            })
+    public Result<Heatmap> createHeatmap(@RequestBody final Heatmap heatmap) throws IOException {
```

Comment:
I'd suggest to use a separate class `HeatmapRegistrationRequest` as input to this method